integration_test.go

```diff
@@ -283,3 +286,89 @@ func TestDefaultValue(t *testing.T) {
 		return result.ID > 0
 	}, "ID should be greater than 0")
 }
+
+func TestDistinct(t *testing.T) {
+	skipIfShort(t)
+	t.Parallel()
+	dsn, cleanup, err := createTestDB(context.Background())
+	if err != nil {
+		log.Fatalf("could not init integration tests while creating database: %v", err)
+		os.Exit(1)
+	}
+	defer cleanup()
+	// Open db.
+	db, err := gorm.Open(New(Config{
+		DriverName: "spanner",
+		DSN:        dsn,
+	}), &gorm.Config{PrepareStmt: true})
+	if err != nil {
+		log.Fatal(err)
+	}
+	if err := db.AutoMigrate(&testutil.User{}, &testutil.Account{}, &testutil.Pet{}, &testutil.Company{}, &testutil.Toy{}, &testutil.Language{},
+		&testutil.Coupon{}, &testutil.CouponProduct{}, &testutil.Order{}, &testutil.Parent{}, &testutil.Child{}); err != nil {
+		t.Fatalf("Failed to migrate models, got error: %v", err)
+	}
+
+	users := []testutil.User{
+		*testutil.GetUser("distinct", testutil.Config{}),
+		*testutil.GetUser("distinct", testutil.Config{}),
+		*testutil.GetUser("distinct", testutil.Config{}),
+		*testutil.GetUser("distinct-2", testutil.Config{}),
+		*testutil.GetUser("distinct-3", testutil.Config{}),
+	}
+	users[0].Age = 20
+
+	if err := db.Create(&users).Error; err != nil {
+		t.Fatalf("errors happened when create users: %v", err)
+	}
+
+	var names []string
```

Comment:
Can we add a comment saying what functionality of `Distinct` are we testing at the beginning of every assertion validation.
For ex: if we are testing Distinct() along with Pluck() we can mention the same.
Here we can add something like `Get name of all the users`.